File: 189-RotateArray/189-RotateArray.cpp
// Last updated: 2/12/2026, 3:45:32 PM
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k %= nums.size();
        vector<int> temp(nums.size());
        for (int i = 0; nums.size() > i; i++) {
            temp[(i + k) % nums.size()] = nums[i];
        }
        nums = temp;
    }
};

File: 189-RotateArray/189-RotateArray.py
# Last updated: 2/12/2026, 3:45:29 PM
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        rotated = nums.copy()
        for i in range(len(nums)):
            rotated[(k + i) % len(nums)] = nums[i]
        nums[:] = rotated